### PR TITLE
Move option validation closer to API layer

### PR DESCRIPTION
**Pull request by aanm**
Please check per commit basis

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5271)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — DISMISSED** [review]

**Comment by ianvernon** [comment]
Tests failed due to an issue in the PR, revoking approval:

```
23:12:27   Cannot set policy enforcement default
23:12:27   Expected command: sudo cilium config PolicyEnforcement=default 
23:12:27   To succeed, but it failed:
23:12:27   Exitcode: 1 
23:12:27   Stdout:
23:12:27    	 
23:12:27   Stderr:
23:12:27    	 Error: Unable to change agent configuration: [PATCH /config][500] patchConfigFailure  Invalid configuration option Unknown option "PolicyTracing"
23:12:27   	 
```

Build link: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5777/

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

File: daemon/daemon.go [hunk 1]
```diff
@@ -1486,11 +1486,18 @@ func (h *patchConfig) Handle(params PatchConfigParams) middleware.Responder {
 
 	d := h.daemon
 
+	cfgSpec := params.Configuration
+
+	om, err := option.DaemonOptionLibrary.ValidateConfigurationMap(cfgSpec.Options)
+	if err != nil {
+		msg := fmt.Errorf("Invalid configuration option %s", err)
+		return api.Error(PatchConfigFailureCode, msg)
```
Comment (raybejjani) [inline daemon/daemon.go]: Should this be `return api.Error(PatchConfigBadRequestCode, err)` ? I think this replaces line 1512/13 (https://github.com/cilium/cilium/pull/5271/commits/7bd52dacd49cd1edcb51d7df2cc6447ef3b260ad#diff-1a1f3e7ad9b1d7584e2d3e7d0c4c3db9R1512) which is also validating?

Relatedly, line 1510 seems to modify the the map so `om` may have fields that should have been removed? It probably doesn't matter, however, but I'm not sure why we removed that option :/

**Review by raybejjani — CHANGES_REQUESTED** [review]

File: daemon/daemon.go (hunk 1 above)
Comment (aanm) [reply daemon/daemon.go]: Ah, good finding

**Comment by aanm** [comment]
test-me-please

**Review by raybejjani — APPROVED** [review]

**Pull request merged**